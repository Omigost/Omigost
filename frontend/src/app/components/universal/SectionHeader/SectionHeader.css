@import "../../../css/Theme.less";

.SectionHeader {
  font-size: 4vw;
  margin-top: 1vw;
  margin-bottom: 1vw;
  display: flex;
  z-index: 999999999;
  position: relative;
  font-family: @PrimaryFont;
}

.TitleWrapper {
  display: inline-block;
  margin-left: 2vw;
  width: 100%;
}

.CenteredTitleWrapper {
  display: inline-block;
  margin-left: 2vw;
  width: 100%;
  text-align: center;
}

.TitleAutolink {
  color: @PrimaryColor;
  text-decoration: none;
  cursor: pointer;
  margin-right: 2vw;
}

.TitleAutolink:hover {
  color: @PrimaryColor;
  text-decoration: none;
  font-weight: bold;
}

.TitleAutolinkInverted {
    color: white;
}

.TitleAutolinkInverted:hover {
    color: white;
}

.IconWrapper {
  display: inline-block;
  width: 45%;
  margin-left: @BigIndent;
}

@media only screen and (max-width: 425px) {
  .SectionHeader {
    font-size: 8vw;
    margin-top: 7vw;
    margin-bottom: 5vw;
  }
}
