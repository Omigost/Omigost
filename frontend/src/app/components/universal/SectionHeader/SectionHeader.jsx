import React from 'react'
import * as styles from './SectionHeader.css'


class SectionHeader extends React.Component {
  render() {
    const { title, centered, sectionId, inverted } = this.props;
    let { autolink } = this.props;

    if(autolink === null || typeof autolink === 'undefined') {
        autolink = true;
    }
    
    return (
      <div styleName='SectionHeader'>
        <div
          styleName={(centered)?('CenteredTitleWrapper'):('TitleWrapper')}
        >
          {
              (autolink)?(
                  <a
                    styleName='TitleAutolink'
                    className={(inverted)?(styles.TitleAutolinkInverted):(null)}
                    href={`#${sectionId}`}
                  >
                    {title}
                  </a>
              ):(title)
          }
        </div>
      </div>
    )
  }
}

export default SectionHeader