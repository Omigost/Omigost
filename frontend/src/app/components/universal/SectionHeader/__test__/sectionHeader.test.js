import React from 'react';
import renderer from 'react-test-renderer';

import SectionHeader from '../SectionHeader.jsx';

test('Snapshot test of SectionHeader', () => {
    const component = renderer.create(
        <SectionHeader
            title='My Section!'
            icon='www.google.com'
            sectionId='mysectionid'
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});