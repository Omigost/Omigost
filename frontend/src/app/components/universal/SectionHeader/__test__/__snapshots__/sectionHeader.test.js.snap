// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of SectionHeader 1`] = `
<div
  styleName="SectionHeader"
>
  <div
    styleName="TitleWrapper"
  >
    <a
      className={null}
      href="#mysectionid"
      styleName="TitleAutolink"
    >
      My Section!
    </a>
  </div>
</div>
`;
